core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -104,6 +129,29 @@ ManifestGroup caseSensitive(boolean newCaseSensitive) {
   }
 
   /**
+   * Returns a iterable of scan tasks. It is safe to add entries of this iterable
+   * to a collection as {@link DataFile} in each {@link FileScanTask} is defensively
+   * copied.
+   * @return a {@link CloseableIterable} of {@link FileScanTask}
+   */
+  public CloseableIterable<FileScanTask> tasks() {
+    Iterable<CloseableIterable<FileScanTask>> tasks = entries((manifest, entries) -> {
+      PartitionSpec spec = specsById.get(manifest.partitionSpecId());
+      String schemaString = SchemaParser.toJson(spec.schema());
+      String specString = PartitionSpecParser.toJson(spec);
+      ResidualEvaluator residuals = ResidualEvaluator.of(spec, dataFilter, caseSensitive);
+      return CloseableIterable.transform(entries, e -> new BaseFileScanTask(
+          e.copy().file(), schemaString, specString, residuals));
+    });
+
+    if (PLAN_SCANS_WITH_WORKER_POOL && manifests.size() > 1) {
```

Comment:
note:`manifest.size() > 1` can be `true` but because of the `manifestPredicate` matching manifest files may be `<= 1` .  I couldn't find a good way to check if the filtered manifests are > 1, but I'd imagine this is more of an optimization so it doesn't hurt in cases where `manifests.size() > 1` but the filtered manifests are <=1